applications/ChimeraApplication/python_scripts/navier_stokes_solver_fractionalstep_chimera.py

```diff
@@ -0,0 +1,113 @@
+from __future__ import print_function, absolute_import, division  # makes KratosMultiphysics backward compatible with python 2.6 and 2.7
+
+# Importing the Kratos Library
+import KratosMultiphysics
+import KratosMultiphysics.ChimeraApplication as KratosChimera
+from KratosMultiphysics.ChimeraApplication import chimera_setup_utils
+
+# Import applications
+import KratosMultiphysics.FluidDynamicsApplication as KratosCFD
+from KratosMultiphysics.FluidDynamicsApplication.navier_stokes_solver_fractionalstep import NavierStokesSolverFractionalStep
+
+
+def CreateSolver(model, custom_settings):
+    return NavierStokesSolverFractionalStepForChimera(model, custom_settings)
+
+class NavierStokesSolverFractionalStepForChimera(NavierStokesSolverFractionalStep):
+
+
+    def __init__(self, model, custom_settings):
+        [self.chimera_settings, self.chimera_internal_parts, custom_settings] = chimera_setup_utils.SeparateAndValidateChimeraSettings(custom_settings)
+        super(NavierStokesSolverFractionalStepForChimera,self).__init__(model,custom_settings)
+        KratosMultiphysics.Logger.PrintInfo("NavierStokesSolverFractionalStepForChimera", "Construction of NavierStokesSolverFractionalStepForChimera finished.")
+
+    def AddVariables(self):
+        super(NavierStokesSolverFractionalStepForChimera,self).AddVariables()
+        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.FLAG_VARIABLE)
+        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.DISTANCE)
+        self.main_model_part.AddNodalSolutionStepVariable(KratosChimera.CHIMERA_DISTANCE)
+        self.main_model_part.AddNodalSolutionStepVariable(KratosChimera.ROTATIONAL_ANGLE)
```

Comment:
are you sure you always need those 4 variables?
I think you only need them for your process that does the rotation right?

Reply:
Yes. Is there a way to add them in the process ?

Reply:
Unfortunately not yet => #859
In the structural solver We have an option in json to do this, but @jcotela did not want to do the same in Fluid...
So up to you
In the long run there will be a better solution

Reply:
Sorry I meant #3561

Reply:
When rotation is not used,  this just adding var to the modelpart and those vars are not used anywhere. So I would say memory overhead is very low. 

I will leave it now like this and adapt the solution in the future !

Reply:
thats the point with historical variables, the memory is allocated whether you use them or not
maybe they can be non-historical?

Up to you if you want to change it